File: backend/src/main/scala/microservices/operations/airline/api/RegisterAirlineManagerPlanner.scala
// RegisterAirlineManagerPlanner 负责operations相关实现。

package com.typesafe.travel.operations.domain

import cats.effect.IO
import com.typesafe.travel.api.routes.ConnectionApiPlan
import com.typesafe.travel.auth.domain.hashPasswordForLoginEmail
import com.typesafe.travel.persistence.operations.AirlineManagerPlainSql
import com.typesafe.travel.shared.kernel.EmailAddress

import java.sql.Connection
import java.time.Instant
import java.util.UUID

object RegisterAirlineManagerPlanner extends ConnectionApiPlan[RegisterAirlineManagerPlannerRequest, ManagerSessionPlannerResponse]:
  override val name: String = "RegisterAirlineManagerPlanner"
  override def plan(input: RegisterAirlineManagerPlannerRequest, connection: Connection): IO[ManagerSessionPlannerResponse] =
    val now = Instant.now()
    val airlineId = s"airline-${UUID.randomUUID().toString.take(12)}"
    val managerId = s"manager-${UUID.randomUUID().toString.take(12)}"
    for
      email <- IO.fromEither(EmailAddress.create(input.email))
      _ <- validateRegisterAirline(input)
      passwordHash <- hashPasswordForLoginEmail(input.password, email)
      _ <- AirlineManagerPlainSql.insertAirline(connection, airlineId, input.airlineName, input.airlineCode, now)
      _ <- AirlineManagerPlainSql.insertAirlineManager(connection, managerId, airlineId, input.email, input.displayName, now)
      _ <- AirlineManagerPlainSql.insertAirlineManagerCredential(connection, managerId, input.email, passwordHash, now)
    yield ManagerSessionPlannerResponse(managerId, "Airline", input.email, input.displayName, "Active", airlineId, None, now.toString)

File: frontend/src/microservices/operations/airline/api/RegisterAirlineManagerPlanner.ts
// 本文件定义 RegisterAirlineManagerPlanner，负责 operations 模块的注册编排和接口入口。

import { executeJsonApiRequest } from '@/shared-kernel/api/ApiTransport'
import type { RegisterAirlineManagerPlannerRequest } from '@/microservices/operations/airline/objects/RegisterAirlineManagerPlannerRequest'
import type { AirlineManagerSessionPlannerResponse } from '@/microservices/operations/airline/objects/AirlineManagerSessionPlannerResponse'

export const registerAirlineManager = (payload: RegisterAirlineManagerPlannerRequest): Promise<AirlineManagerSessionPlannerResponse> =>
  executeJsonApiRequest<AirlineManagerSessionPlannerResponse>('/RegisterAirlineManagerPlanner', 'POST', payload)
